stacker_blueprints/firehose.py

```diff
@@ -0,0 +1,197 @@
+
+    statements = [
+        Statement(
+            Effect=Allow,
+            Action=[
+                awacs.s3.AbortMultipartUpload,
+                awacs.s3.GetBucketLocation,
+                awacs.s3.GetObject,
+                awacs.s3.ListBucket,
+                awacs.s3.ListBucketMultipartUploads,
+                awacs.s3.PutObject,
+            ],
+            Resource=[
+                s3_arn(bucket),
+                s3_arn(Join('/', [bucket, '*'])),
+            ],
+        ),
+    ]
+    return Policy(Statement=statements)
+
+
+class Firehose(Blueprint):
+
+    PARAMETERS = {
+        'Role': {
+            'type': 'String',
+            'description': 'The role that should have access to write to firehose.',
+            'default': '',
+        },
+        'BucketName': {
+            'type': 'String',
+            'description': 'Name for the S3 Bucket',
+        },
+    }
+
+    def create_bucket(self):
+        t = self.template
+        t.add_resource(
+            s3.Bucket(
+                BUCKET,
+                BucketName=Ref('BucketName'),
+            )
+        )
+        t.add_output(Output('Bucket', Value=Ref(BUCKET)))
+
+    def generate_iam_policies(self):
+        ns = self.context.namespace
+        s3_policy = iam.Policy(
+            S3_WRITE_POLICY,
+            PolicyName='{}-s3-write'.format(ns),
+            PolicyDocument=s3_write_policy(Ref('BucketName')),
+        )
+        logs_policy = iam.Policy(
+            LOGS_WRITE_POLICY,
+            PolicyName='{}-logs-write'.format(ns),
+            PolicyDocument=logs_write_policy(),
+        )
+        return [s3_policy, logs_policy]
+
+    def create_role(self):
+        t = self.template
+
+        statements = [
+            Statement(
+                Principal=Principal('Service', ['firehose.amazonaws.com']),
+                Effect=Allow,
+                Action=[sts.AssumeRole],
+                Condition=Condition(
+                    StringEquals('sts:ExternalId', Ref('AWS::AccountId')),
+                ),
+            ),
+        ]
+        firehose_role_policy = Policy(Statement=statements)
+        t.add_resource(
+            iam.Role(
+                IAM_ROLE,
+                AssumeRolePolicyDocument=firehose_role_policy,
+                Path='/',
+                Policies=self.generate_iam_policies(),
+            ),
+        )
+        t.add_output(Output('Role', Value=Ref(IAM_ROLE)))
+        t.add_output(Output('RoleArn', Value=GetAtt(IAM_ROLE, 'Arn')))
+
+    def create_policy(self):
+        ns = self.context.namespace
+        t = self.template
+
+        t.add_condition(
+            'ExternalRole',
+            Not(Equals(Ref('Role'), '')),
+        )
+
+        t.add_resource(
+            iam.PolicyType(
+                FIREHOSE_WRITE_POLICY,
+                PolicyName='{}-firehose'.format(ns),
+                PolicyDocument=firehose_write_policy(),
+                Roles=[Ref('Role')],
```

Comment:
I think I have a misunderstanding here - we create a role in `create_role` but we don't use it here?